backend/src/apiserver/storage/resource_reference_store.go

```diff
@@ -176,6 +187,6 @@ func (s *ResourceReferenceStore) scanRows(r *sql.Rows) ([]model.ResourceReferenc
 	return references, nil
 }
 
-func NewResourceReferenceStore(db *DB) *ResourceReferenceStore {
-	return &ResourceReferenceStore{db: db}
+func NewResourceReferenceStore(db *DB, pipelineStore PipelineStoreInterface) *ResourceReferenceStore {
```

Comment:
Could you add a comment explaining that if the pipeline store is specified, it will be used to query pipelines and pipeline versions rather than direct database queries.